lib/IRGen/GenClangDecl.cpp

```diff
@@ -80,7 +80,7 @@ void IRGenModule::emitClangDecl(const clang::Decl *decl) {
     // If it's a member of a file-level decl, like a C++ static member variable,
     // we want to add the entire file-level declaration because Clang doesn't
     // expect to see members directly here.
-    for (auto *DC = D->getDeclContext();; DC = DC->getParent()) {
+    for (auto *DC = D->getLexicalDeclContext();; DC = DC->getParent()) {
```

Comment:
I'm not sure this is right; your logic makes us ignore local externs, which works in your test case only because we don't actually need to emit the variable/function.  We need to look for a non-local-extern declaration.

Reply:
Can you expand a little more about that? I'm not sure I follow 100%. In practice, there are cases where the resolution of the local extern may not happen until link time.

Reply:
Sure, and in those cases we aren't responsible for emitting the declaration; it's the programmer's responsibility to ensure that there's a matching definition  But if there *is* a definition in the header, it might still be our obligation, e.g.

```
int get_x() {
  extern const int x;
  return x;
}
const int x = 5;
```

Reply:
I see the problem: as it stands, we're not emitting any references to symbols from local externs, even when we are able to resolve them immediately. I'll look in to an alternative way of handling this.